### PR TITLE
infrastructure as a code for badge-poser deployment on aws ec2

### PR DESCRIPTION
**Pull request by fabiocicerchia**

### CODE REVIEW

**Comment by JellyBellyDev** [comment]
Thanks a lot @fabiocicerchia, but why you put the files in `/config/sys/*`?
The `/config/` dir is reserved for the framework. no?
Wouldn't it be better to put all this files in a separate dir from the name `deploy` in the root of project?

**Comment by garak** [comment]
I personally agree with Fabio: `config` dir is meant to contains every project configuration, not just Symfony ones

**Comment by JellyBellyDev** [comment]
Could you configure these settings during the provisioning phase?
https://symfony.com/doc/current/performance.html#configure-opcache-for-maximum-performance
https://symfony.com/doc/current/performance.html#don-t-check-php-files-timestamps
https://symfony.com/doc/current/performance.html#configure-the-php-realpath-cache

And after switch new version please use this tool for the reset opcache: https://github.com/gordalina/cachetool with the following command:
`php cachetool.phar opcache:reset`

Thnaks

File: config/sys/playbooks/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    # ############
+
+    - name: "Install App"
+      shell: |
+        cd /application/releases/{{ deploy_time }}
+        cp /application/.env.dist /application/releases/{{ deploy_time }}.env
+
+        wget https://raw.githubusercontent.com/composer/getcomposer.org/76a7060ccb93902cd7576b67264ad91c8a2700e2/web/installer -O - -q | php -- --quiet
+
+        php composer.phar install
```
Comment (JellyBellyDev) [inline config/sys/playbooks/deploy.yml:48]: `composer install --no-ansi --no-dev --no-interaction --no-plugins --no-progress --no-scripts --no-suggest --optimize-autoloader`?

File: config/sys/playbooks/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        yarn install
```
Comment (JellyBellyDev) [inline config/sys/playbooks/deploy.yml:49]: `yarn install --production`?

File: config/sys/playbooks/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        yarn dev
```
Comment (JellyBellyDev) [inline config/sys/playbooks/deploy.yml:50]: `yarn build`?

File: config/sys/playbooks/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    # RELEASE
+    # ########
+
+    - pause:
+        prompt: "Do you want to release the version {{ deploy_time }}?"
+
+    - name: "Release"
+      shell: |
+        ln -s /application/releases/{{ deploy_time }} /application/new
+        mv -T /application/new /application/current
```
Comment (JellyBellyDev) [inline config/sys/playbooks/deploy.yml:61]: For this task and others above you should use the `deploy_helper` of ansible: https://docs.ansible.com/ansible/latest/modules/deploy_helper_module.html

File: config/sys/playbooks/setup.yml [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+- name: "Provision EC2"
+  hosts: "web-1"
+  become: true
+
+  # VARS
+  # ############################################################################
+
+  vars:
+    domain: poser.pugx.org
+    email: pugx@example.com
```
Comment (JellyBellyDev) [inline config/sys/playbooks/setup.yml:11]: `example.com`? @garak we have an email address for the `pugx.org`?

Comment (garak) [reply config/sys/playbooks/setup.yml:11]: I'm afraid not 😔

Comment (JellyBellyDev) [reply config/sys/playbooks/setup.yml:11]: `pugroma at gmail.com`?

Comment (garak) [reply config/sys/playbooks/setup.yml:11]: That's an email we're using for PUG Roma twitter account.
If you think it's suitable here, feel free to use it

Comment (fabiocicerchia) [reply config/sys/playbooks/setup.yml]: done

File: config/sys/playbooks/deploy.yml (hunk 4 above)
Comment (fabiocicerchia) [reply config/sys/playbooks/deploy.yml]: done

File: config/sys/playbooks/deploy.yml (hunk 3 above)
Comment (fabiocicerchia) [reply config/sys/playbooks/deploy.yml]: done

File: config/sys/playbooks/deploy.yml (hunk 2 above)
Comment (fabiocicerchia) [reply config/sys/playbooks/deploy.yml]: done

File: config/sys/playbooks/deploy.yml (hunk 1 above)
Comment (fabiocicerchia) [reply config/sys/playbooks/deploy.yml]: done

File: config/sys/playbooks/deploy.yml [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+        yarn install --production
+
+        /usr/local/bin/composer require encore
+        yarn add @symfony/webpack-encore --production
+        yarn add sass-loader@^8.0.0 node-sass --dev
+
+        yarn build
+
+        chmod -R 777 var/cache var/log
```
Comment (JellyBellyDev) [inline config/sys/playbooks/deploy.yml:69]: You can override this instruction with:
```
- name: Setfacl var folder permissions
  shell: HTTPDUSER=`ps axo user,comm | grep -E '[a]pache|[h]ttpd|[_]www|[w]ww-data|[n]ginx' | grep -v root | head -1 | cut -d ' ' -f1` &&
    setfacl -dR -m u:"$HTTPDUSER":rwX -m u:$(whoami):rwX {{ deploy_helper.new_release_path }}/var &&
    setfacl -R -m u:"$HTTPDUSER":rwX -m u:$(whoami):rwX {{ deploy_helper.new_release_path }}/var
```

File: config/sys/playbooks/deploy.yml [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+    - pause:
+        prompt: "Do you want to release the version {{ deploy_helper.new_release_path }}?"
+
+    - name: "Release"
+      deploy_helper:
+        path: /application
+        release: "{{ deploy_helper.new_release }}"
+        state: finalize
+        keep_releases: 10
```
Comment (JellyBellyDev) [inline config/sys/playbooks/deploy.yml:82]: 3?

File: config/sys/playbooks/deploy.yml [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+      notify:
+      - reload nginx
+
+    - name: "Reset opcache"
+      shell: |
+        curl -sO https://gordalina.github.io/cachetool/downloads/cachetool.phar
+        chmod +x cachetool.phar
+        php cachetool.phar opcache:reset --fcgi
+        rm cachetool.phar
+
```
Comment (JellyBellyDev) [inline config/sys/playbooks/deploy.yml:92]: Where is the command to update contributors?

File: config/sys/playbooks/deploy.yml (hunk 6 above)
Comment (fabiocicerchia) [reply config/sys/playbooks/deploy.yml]: done

File: config/sys/playbooks/deploy.yml (hunk 8 above)
Comment (fabiocicerchia) [reply config/sys/playbooks/deploy.yml:101]: added

File: config/sys/playbooks/deploy.yml (hunk 7 above)
Comment (fabiocicerchia) [reply config/sys/playbooks/deploy.yml]: done

**Pull request merged**